src/goto-instrument/cover.cpp

```diff
@@ -1553,7 +1553,10 @@ void instrument_cover_goals(
         remove_repetition(controlling);
         // for now, we restrict to the case of a single ''decision'';
         // however, this is not true, e.g., ''? :'' operator.
-        minimize_mcdc_controlling(controlling, *decisions.begin());
+        if(!decisions.empty())
```

Comment:
I believe this should be expressed via `CHECK_RESULT` or `PRECONDITION` as empty decisions would seem wrong at this point.

Reply:
I suggest this is fixed by someone with more domain knowledge. I made this change because decisions *isn't* always populated, but finding out why it's sometimes empty goes beyond the scope of a simple memory-bug fix. Changing this to an INVARIANT will cause some of the tests to fail, which in turn will prevent this PR being merged.

Reply:
I'd suggest the following:
1. Use `git blame` to identify the author of those lines.
2. Use (some variation of) `INVARIANT`.
3. Mark all failing tests as `KNOWNBUG`.
4. File an issue and assign it to the author identified in 1.